semantic_release/cli/commands/changelog.py

```diff
@@ -62,10 +52,55 @@ def changelog(ctx: click.Context, release_tag: str | None = None) -> None:
             )
             ctx.exit(0)
 
-        changelog_text = render_default_changelog_file(env)
-        with open(str(changelog_file), "w+", encoding="utf-8") as f:
-            f.write(changelog_text)
+        with open(str(changelog_file), "a+", encoding="utf-8") as f:
+            f.seek(0)
+            existed_changelog_text = f.read()
+            f.truncate()
+            existed_changelog_sections = existed_changelog_text.split(CHANGELOG_PLACEHOLDER)
+            if len(existed_changelog_sections) > 1:
```

Comment:
I think it's a much simpler check to have `f"{CHANGELOG_PLACEHOLDER}\n" in f.readlines()` - avoids trying to split. The text could (by chance) by in a commit message which would lead to issues.

You can always do

```python3
existing_changelog_text = f.read()
f.seek(0)
lines = f.readlines()
```
to get both, or even
```python3
lines = f.readlines()
existing_changelog_text = "".join(lines)
```